tests/integration_tests/tests/agentless_tests/test_task_retries.py

```diff
@@ -127,16 +127,34 @@ def assertion():
                     break
             else:
                 self.fail('could not find expected message')
-            events = [e for e in events if
-                      e.get('event_type', '').startswith('task_') or
-                      e.get('event_type', '') == 'sending_task']
-            seen_current_retries = set()
-            for event in events:
-                current_retries = event['context']['task_current_retries']
-                seen_current_retries.add(current_retries)
-                total_retries = event['context']['task_total_retries']
-                self.assertEqual(total_retries, 5)
-            self.assertSetEqual(seen_current_retries, {0, 1, 2, 3})
+
+            # We're looking only at the events from the create operation
+            retry_events = [e for e in events if e['context']['operation'] ==
+                            'cloudify.interfaces.lifecycle.create']
+            retry_events = sorted(retry_events, key=lambda e: e['timestamp'])
+            self.assertTrue(len(retry_events), 12)
+
+            retries = 0
+            for i in range(0, 10, 3):  # Set the range's step to 3
```

Comment:
btw can also do `for send_task, start_task, reschedule_task in zip(retry_events[::3], retry_events[1::3], retry_events[2::3])` and skip using the index altogether